### DISCUSSION TITLE
New UI Source Code

### DISCUSSION THREAD

**Discussion by CodingPupper3033** [inventree/InvenTree#6402] [General]
Hello!
I have heard that you guys are hard at work on a new UI overhaul. Where would I be able to see the source code for the new UI? I haven't been able to find it.

Also, I understand it's built upon React, but is it also still using Django and Bootstrap, or is it being completely overhauled?

Thanks for your time!

**Comment by SchrodingersGat** [comment]
Hey @CodingPupper3033 

> Where would I be able to see the source code for the new UI?

You can find the source here: https://github.com/inventree/InvenTree/tree/master/src/frontend

> Also, I understand it's built upon React, but is it also still using Django and Bootstrap, or is it being completely overhauled?

The frontend is being reworked to use react, but the backend is still django / python. There will not be anymore use of bootstrap.

- You can read more about the overhaul in [this blog post](https://inventree.org/blog/2023/08/28/react)
- See the new interface in action - https://demo.inventree.org/platform/

**Comment by CodingPupper3033** [comment]
Thank you, that is very helpful!

Would I be able to get a list of React libraries and setup being used?

I would like to contribute, but I should tinker with the libraries and frameworks before I attempt to help.

**Reply by SchrodingersGat** [reply]
Contributing guide for the frontend is here - https://docs.inventree.org/en/latest/develop/react-frontend/

The libraries being used are all discoverable in `/src/frontend/package.json`

**Reply by CodingPupper3033** [reply]
Thank you!
